stun/discover.go

```diff
@@ -164,3 +164,93 @@ func (c *Client) discover(conn net.PacketConn, addr *net.UDPAddr) (NATType, *Hos
 	}
 	return NATSymmetric, mappedAddr, nil
 }
+
+func (c *Client) behaviorTest(conn net.PacketConn, addr *net.UDPAddr) (*NATBehavior, error) {
+	natBehavior := &NATBehavior{}
+
+	// Test1   ->(IP1,port1)
+	// Perform test to check if it is under NAT.
+	c.logger.Debugln("Do Test1")
+	resp1, err := c.test(conn, addr)
+	if err != nil {
+		return nil, err
+	}
+	if resp1 == nil {
+		return nil, errors.New("NAT blocked.")
+	}
+	// identical used to check if it is open Internet or not.
+	if resp1.identical {
+		return nil, errors.New("Not behind a NAT.")
+	}
+	// use otherAddr or changedAddr
+	otherAddr := resp1.otherAddr
+	if otherAddr == nil {
+		if resp1.changedAddr != nil {
+			otherAddr = resp1.changedAddr
+		} else {
+			return nil, errors.New("Server error: no other address and changed address.")
+		}
+	}
+
+	// Test2   ->(IP2,port1)
+	// Perform test to see if mapping to the same IP and port when
+	// send to another IP.
+	c.logger.Debugln("Do Test2")
+	tmpAddr := &net.UDPAddr{IP: net.ParseIP(otherAddr.IP()), Port: addr.Port}
+	resp2, err := c.test(conn, tmpAddr)
+	if err != nil {
+		return nil, err
+	}
+	if resp2.mappedAddr.IP() == resp1.mappedAddr.IP() &&
```

Comment:
got the following error here
```
➜  go-stun git:(test) ./go-stun -b
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x110ead2]

goroutine 1 [running]:
github.com/ccding/go-stun/stun.(*Client).behaviorTest(0xc000159ef8, 0x117b480, 0xc00012a028, 0xc000114780, 0x0, 0x0, 0x0)
        /Users/cong/go/src/github.com/ccding/go-stun/stun/discover.go:204 +0x232
github.com/ccding/go-stun/stun.(*Client).BehaviorTest(0xc00003eef8, 0x0, 0x0, 0x0)
        /Users/cong/go/src/github.com/ccding/go-stun/stun/client.go:120 +0xe5
main.behaviorTest(0xc00003eef8)
        /Users/cong/go/src/github.com/ccding/go-stun/main.go:61 +0x45
main.main()
        /Users/cong/go/src/github.com/ccding/go-stun/main.go:41 +0x6a5
```

Reply:
Also `local: 62.24.197.181:1298` is not my local IP address

Reply:
It should be nat blocked , but will be detected in the first step.Maybe the network is not stable.

Reply:
> Also `local: 62.24.197.181:1298` is not my local IP address

Is it better to use  mapped   instead of  local?